doc/source/advanced.md

```diff
@@ -199,3 +199,50 @@ To explicitly specify this strategy, use the following in your `config.yaml`:
 singleuser:
     schedulerStrategy: pack
 ```
+
+## Pre-pulling Images for Faster Startup
+
+When user images are of non-trivial sizes (as most of them are), **pre-pulling**
+the images on all nodes significantly decreases the amount of time it takes to
+start a user's server. Without pre-pulling, sometimes it can take upto 5-10
+minutes for a user's server to launch. With pre-pulling, it is cut down to a few
+seconds in most cases.
+
+By default, the user container image is pulled on to all nodes whenever a `helm
+install` or `helm upgrade` is performed. This causes `helm install` and `helm
+upgrade` to take several minutes, but mostly that is worth it since it makes
+user startup performance much faster. By default `helm install` or `helm
```

Comment:
New paragraph:

With the default **pre-pulling** setting, a `helm install` or `helm upgrade` will cause the system to wait for 5 minutes to begin pulling the images before timing out. This wait time is configurable by passing the `--wait <seconds>` flag to the `helm` commands.